src/main/scala/cromwell/webservice/WorkflowQueryKey.scala

```diff
@@ -0,0 +1,94 @@
+package cromwell.webservice
+
+import org.joda.time.DateTime
+
+import scala.language.{postfixOps, reflectiveCalls}
+import scala.util.{Success, Try}
+import scalaz.Scalaz._
+import scalaz.{Success => _, _}
+
+object WorkflowQueryKey {
+  val ValidKeys = Set(StartDate, EndDate, Name, Status) map { _.name }
+
+  case object StartDate extends DateTimeWorkflowQueryKey {
+    override val name = "Start"
+    override def displayName = "start date"
+  }
+
+  case object EndDate extends DateTimeWorkflowQueryKey {
+    override val name = "End"
+    override def displayName = "end date"
+  }
+
+  case object Name extends SeqStringWorkflowQueryKey {
+    override val name = "Name"
+    private val WorkflowNamePattern = "([a-zA-Z][a-zA-Z0-9_]+)".r
+
+    override def validate(grouped: Map[String, Seq[(String, String)]]): ValidationNel[String, Seq[String]] = {
+      val values = valuesFromMap(grouped).toList
+      val nels = values map {
+        case WorkflowNamePattern(n) => n.successNel
+        case v => v.failureNel
+      }
+      sequenceListOfValidationNels("Name values do not match allowed workflow naming pattern [a-zA-Z][a-zA-Z0-9_]+", nels)
+    }
+  }
+
+  case object Status extends SeqStringWorkflowQueryKey {
+    override val name = "Status"
+    val ValidStatuses = Set(
+      "Submitted",
+      "Running",
+      "Aborting",
+      "Failed",
+      "Succeeded",
+      "Aborted"
+    )
+
+    override def validate(grouped: Map[String, Seq[(String, String)]]): ValidationNel[String, Seq[String]] = {
+      val values = valuesFromMap(grouped).toList
+      val nels = values map { v =>
+        if (ValidStatuses.contains(v.toLowerCase.capitalize)) v.successNel else v.failureNel
```

Comment:
Maybe use `WorkflowState.fromString` here instead of duplicating the valid statuses ?
`if (Try(WorkflowState.fromString(v.toLowerCase.capitalize)).isSuccess)`